unison-cli/src/Unison/Codebase/Editor/VersionParser.hs

```diff
@@ -28,13 +28,16 @@ import qualified Unison.Codebase.Path as Path
 defaultBaseLib :: Parsec Void Text ReadShareRemoteNamespace
 defaultBaseLib = fmap makeNS $ release <|> unknown
   where
-    unknown, release, version :: Parsec Void Text Text
+    unknown, release, milestoneVersion :: Parsec Void Text Text
     unknown = pure "main" <* takeWhileP Nothing (const True) <* eof
-    release = fmap ("releases." <>) $ "release/" *> version <* eof
-    version = do
-      v <- Text.pack <$> some (alphaNumChar <|> ('_' <$ oneOf ['.', '_', '-']))
+    release = fmap ("releases." <>) $ "release/" *> milestoneVersion <* eof
+    -- Parses the milestone of the current version; e.g. M4a -> M4
+    milestoneVersion = do
+      m <- char 'M'
```

Comment:
I don't love hard-coding the `'M'` here. What's the fallback if it's not there? Or just remember to fix it later?

Reply:
How about look specifically for `'M'` <numbers> <optional letters>, and parse it apart only in that case; otherwise use the whole thing like before?

Reply:
We already pull from `main` in the fallback case, I don't necessarily understand what you're suggesting here. Are you just worried that we'll change our versioning scheme in the future?

If so, then yes, that's true, but I'd rather fall-back to `main` and have to come back and change this later than try pulling a release that doesn't exist 🤔